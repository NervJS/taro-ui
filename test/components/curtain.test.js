import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtCurtain from '../../.temp/components/curtain/index'

describe('AtCurtain Snap', () => {
  it('render initial AtCurtain', () => {
    const component = renderToString(<AtCurtain />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCurtain -- props className', () => {
    const component = renderToString(<AtCurtain className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCurtain -- props customStyle', () => {
    const component = renderToString(<AtCurtain customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCurtain -- props isOpened', () => {
    const component0 = renderToString(<AtCurtain isOpened> test </AtCurtain>)
    expect(component0).toMatchSnapshot()
    const component1 = renderToString(
      <AtCurtain isOpened={false}> test </AtCurtain>
    )
    expect(component1).toMatchSnapshot()
  })

  it('render AtCurtain -- props closeBtnPosition', () => {
    const component0 = renderToString(
      <AtCurtain closeBtnPosition='top'> test </AtCurtain>
    )
    expect(component0).toMatchSnapshot()
    const component1 = renderToString(
      <AtCurtain closeBtnPosition='bottoms'> test </AtCurtain>
    )
    expect(component1).toMatchSnapshot()
  })
})
