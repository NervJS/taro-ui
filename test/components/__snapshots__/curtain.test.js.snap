// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCurtain Snap render AtCurtain -- props className 1`] = `"<div class=\\"at-curtain at-curtain--closed test\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"><div class=\\"at-curtain__btn-close at-curtain__btn-close--bottom\\"></div></div></div></div>"`;

exports[`AtCurtain Snap render AtCurtain -- props closeBtnPosition 1`] = `"<div class=\\"at-curtain at-curtain--closed\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"> test <div class=\\"at-curtain__btn-close at-curtain__btn-close--top\\"></div></div></div></div>"`;

exports[`AtCurtain Snap render AtCurtain -- props closeBtnPosition 2`] = `"<div class=\\"at-curtain at-curtain--closed\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"> test <div class=\\"at-curtain__btn-close at-curtain__btn-close--bottoms\\"></div></div></div></div>"`;

exports[`AtCurtain Snap render AtCurtain -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-curtain at-curtain--closed\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"><div class=\\"at-curtain__btn-close at-curtain__btn-close--bottom\\"></div></div></div></div>"`;

exports[`AtCurtain Snap render AtCurtain -- props isOpened 1`] = `"<div class=\\"at-curtain\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"> test <div class=\\"at-curtain__btn-close at-curtain__btn-close--bottom\\"></div></div></div></div>"`;

exports[`AtCurtain Snap render AtCurtain -- props isOpened 2`] = `"<div class=\\"at-curtain at-curtain--closed\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"> test <div class=\\"at-curtain__btn-close at-curtain__btn-close--bottom\\"></div></div></div></div>"`;

exports[`AtCurtain Snap render initial AtCurtain 1`] = `"<div class=\\"at-curtain at-curtain--closed\\"><div class=\\"at-curtain__container\\"><div class=\\"at-curtain__body\\"><div class=\\"at-curtain__btn-close at-curtain__btn-close--bottom\\"></div></div></div></div>"`;
